### DISCUSSION TITLE
What is the default Time view (e.g. the time unit)?

### DISCUSSION THREAD

**Discussion by colmanhumphrey** [vega/vega-lite#8216] [General]
When testing out plotting things over time, the default view when `timeUnit` is unspecified is often unclear to me. It often seems to use seconds, but creates "fat" bars, e.g. [here](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUMSSCGcCGgDaoZZgD6mAF5oQAJgAMt2-3sA2frYDsAFSeoALPao9gCcAFqKIAyyNACOemY0yjYwAB4A8gAyACIwAGphTADWYQCCmQCyAOoAcgDusqXKMt40AJphbIJ1ABoIlhUVRW0gAL4KJuZWNg62AMwuYh623vYAjKhiQXNiEUrRcQlJNjUAChUA4rGWmQBOpQCSAKIQxGEAQlUAYtl1RQBKxAqT1KOGybTYvSgAE8wl8HmMJioptZ0LMFvYlg5fLZUGt-MEABx7KIxeJwRLJdBhKrEUoIADKpwAZrFZNCimQHkxTnV3gg2AwnnliNDSgBFC49KA1aE0dIMgzjSYWVF2RzOUL8NYhVZiYIbfy7SIHcmUmypfJsApQCqCBkIAASpC+kneH3FbTImFkDKK6ViAClA8QANLZJjinCIlXTNGONYudz2QmrDZiXFrTwk01HKkgCrvL4XTLKACqEEyXzI6R6ZbCD1SFyK72ZpzyeUwEFk2VKDFDpSQqVD-xjyNVMwTLkJLjWOIChNQc3CJrJeZsZG8pS+PVuzK+zJognemBqZBg6VuX0wNB6gbg3ggCEE-0syh62TIXrHpgn8aciwuLqtiGnMARzDma4Usc6ClGwD6pDgCBFAwNRwIIbRMO87x1NkoY1E8mQ0EgcDZN4sj-A8waFhcIzKuOcbqvMs4uD4jh4vqcwQauhzQfmRTKIWTBVMosinKkFSpKkUCSDQTwAMJsPJNQPEg-zKDcdTKNcMA9DAjpPD+KKTiBLjOGsqY6gE6bEjxZowSANCSEw-zXNC2RIKGxAPJYlhQOKkh1N5sRwBcZD-NCTDQg8CBPP0sQ1EwRl-kxiZrM4tj+N46UBEERqQbx5roLg8kMvJRQQP8OC3HAcCBsg6TEE8FTZD0NBwJYpRtHATCOvumRkGQDCYMljHoi4-jallmIBCEqDzAV9n5lAzIMAyDAMFpZbQuksiCGWSC3I6mA7sQmAPLQmBwNySBFPAsRPFUo1quNmLamIvgbKES6uIt67oDgOBhOkdTpDgPQbXkFRkDUmTiggtzZMoFy3E8bAIAwNCnKGTxlt4mTEKhz0mYmb1zPYn3BAS-i2fsUFFSA4pwF24YPFUYR4AgTAXMopQ1LceQMuK1aOjQjqBlAFw4MQbDKIIUB1MT-6kyEizZa4wQawtdn-SArLKWWbxkFGsjikwtzeD0OBIEK8lfNCbRPLEyjpPk8kwPJNCvErqWAeTvhgW4eJrH9fE2DSWBPIG1URpg3hBQ0si8xUeTpMUpT-N4RRVGwsSpOQPw+69SxuKsQQppstihwzTxFEgEChnkODQjgVSWJgVQPHAeQQF8tzKHAYQ9A82QQOKjo4AwganGwpzybcRdTmsib+yBS4Eo41cOU8bQIBUyg4IeDynFUqRkBUCA4JkUDZNbdxIDAmQ9MoeS0jg3htNpi-MW9tiuKsGt7CeCXNxOmhUHJwAaEWB4FRbhVD0BcBkygGA9DIGESwsRxTDTLEUCoMIzzvFUKkPq39NSeGnKsYBYhDTZh1mHdASAmCCDIDgJgFQ+zvCeI6JgXxiAwAqtPUoOMoBtBwKFQMTxLxwFpGwUhLF-ZZnmhrfwoDSTgPzLEf4jpxSyCDPJfatwLh1CQGWWQmRUitmhFotgZZSgVBwDAKobQdAwGUJIORWoUyUOCEEbWYClo2FKIIbICU2ilAdE8f4ZxAxIAeIlMIZZxSpHgYGdI2RBAIGyP8eScAkBQFON-DEaU5jZTmmsMCaxjT+N1oGbqukOoIG8N0UMjo8ihneDUKArRoTQieJYWQMtQyDyKDUQQFwGAwG-qTJYax-5WXmPNWh1T6EgDyMoTItZZCSDDLIWQbQ6jQlqekBGDwGDMjLJYbO75LB1DgKUS8KcPGTUcNlL6YExBLLUQE9AVQcCxEyBcbI4zemcizjAMsDBxT-C+KkTAZALhR3lDASQTwk7igeEUKZE0PA+FmcEYBtgqlfN1hcOkmE2ikGIKJW4EAD6SHkoGPIFVoCWBqGWR0qQywIDaGWPIBz3H0V-GNKcQDtS4oJFxeaRLcwrJwF8L4DBTgDK2byU46RmTineLEYgtx0iWCYKGC4pRYDYQgBUKMhilRIiFS9RwGJnn-wWesPE9gt75n+KUOokgZ6WDgDgf4oZ5IGqQY2IoFxAwMDLKGWemAsEXFlpISQZAGRyOXkmbKwCjSbBXMshmEJxRljaDFBktwEBfDCJkR0khoRsBoORcNlhxQdneIGEsZYKgQHklABAWK3DahKWvMQ7zfp0IZsod4NwyCOjLDFLJ6Q2iBnZk0WQ3hQypD2eKbw8DSjvDIGwKAYRTjQixWmrwqwynANCG6mwFRQw0gQK-HcTBZCWDCKKVImQajpCeDQYgqQngMnrJIOATwslkGZF3DxaVcXKIJCBa96AHihliAgLBVQ911AQKUPIZYXiBgeCPL4uTMbvBoGWSQmBemnCQJkeShTsVrApoovw9hXWjocnkbIr93hd3eI6eWFKijEEyAgWQDx-gMCgCM04ZBsiOnSF5HoEJeFQe1HMrWGxCUIZAIynmxBTiWGhJyuAzIcCyFYd4GAghoRVC5jAAiygaC4IYF8HAUNMWCuMv+TUk0-7ZX1NQvEVd2P5hueKUMkgb5FG8DUCoYRAzmtOP8PIDw3ROeUFUBksgqjyXKKoZ4dFrVed9mmxj2VFwqKXJ8mVDNXEVCYNkCo6R0alCYMyR0Xw2HXwwZkQQRQHhsAeG0WSpxTiYEdAcsgvbNQfXYl4Di2nYjYY6922IbAIBvguBUYgUAyDQkyK+uADJMihjIJYHAZ5KzEE5IrTzKVZiahnBZVYYF-DKNUTVhyFx5PPsdP8No0NJBVFXf8BAzkX4QAeLcNo6QHj2liqeF+sQnp3eFQBN6mJfBBDEASRj2mDzeHSGLa47w9llluEpQQkhZDyVSEUJgbRBD7uUKDFtDxmSWFuJM1HtrTKeLnAssQi4MzacwD0TAec6hsus0UTncAV0IDLOkGoOrMiyCgGpcUtxPSdLgMobwvaS5zkY5sam-htPA8DGQJ0jooBsHFnD1lw22iQ1qJkFkUm1n7wqKcKU9H7ALDPQsuYG8Q4hZsFUGgXx5IMGII6OADB5JOzIE8Kin5Yh5G+wgKJSAvhtFuLIHooZF1117TOWwqZvo7HmsF3NDl0UwCgLcFavcwXyR6GwJgdQcCFnUCPOoB0kAIAeHUL4CASwLx5yZcyGUVh4sq+lbToN2YNwuOkCZ0A9JkFKFge+gY3TBOZMQWISD5JkCqJ6k9s+0zBDmv4c34fEMRskBpGAlgnioxqOKQQecej-HSO8LyjhMyFxgPpYD0IIMQNkFUIZFPt5kmNqCsPYLjmBHMLXsSishAN6HkJ+neggHAG0PJPJMyP8IGAyIIMoMyAyLRFUMyLIH1JChAN4DhBUJkGMAALrjAgB1BJD0BoCuCsZKDwA0DngjS5T2BKA2yYDrY0C+poCgCUZiI2CHgjRKB26yDqA0CyAGAAxIDKCOayBTZcHWy3BD46GGDsFKDIC3CYqaAgCKFwA2BMCHSRDSFdA3jRh2HqFXQaE2Dy524GFGFcH+GyxaFTZ2GpDyF6ztQUE2A2quEtyOHFSIA4BsCmGCCIggDHp2GHjoT5ifbJBKAOE2CyDoxaE6CRBICDQ1QUBXRxFew3jq4jSjAtFAA):
<img width="611" alt="CleanShot 2022-06-08 at 13 07 31@2x" src="https://user-images.githubusercontent.com/6945891/172675924-a211345d-e580-488b-b799-9902e063b149.png">

In contrast, if you actually set the unit to be say `yearmonthdatehoursminutesseconds`, you get the same plot but with skinny lines:
<img width="616" alt="CleanShot 2022-06-08 at 13 07 23@2x" src="https://user-images.githubusercontent.com/6945891/172675903-bd4e2120-ae4e-42cf-9564-6e77915196ee.png">

I'm guessing some combination of bar width etc can recreate the first, but maybe then the question will become: why does the default view not use the default step/bin size?

Thanks!

**Comment by domoritz** [comment]
The first chart doesn't use a time unit at all and just places bars at the timestamp and then draws them with a particular pixel width (5). You can see that in the compiled Vega.

**Reply by colmanhumphrey** [reply]
Is that intended behavior? I guess I am surprised that the default behavior from specifying `"type": "temporal"` is to not use a time unit (alternatively: that the default is not simple to replicate with manual specifications). From another perspective, I guess maybe I just didn't expect this as default behavior, I'm not sure when I would choose this as a view for time.

**Reply by kanitw** [reply]
It's probably useful to specify what you expect given that you didn't specify any time transform to the field?

**Reply by colmanhumphrey** [reply]
Hey @kanitw thanks for replying! The context here is helping others who may not be super familiar with the spec put graphs together and put them in the "pit of success" (otherwise yeah, generally a time transform would make a lot of sense).

I would be happy to take a guess at what I'd expect, but (and I could be wrong) I think that might be worth having a different discussion for that. That is, I posted this discussion not because I think "this default isn't what I expected" but rather "I don't think I understand this default". That is, there are plenty of reasonable defaults (and I probably only understand a subset of those), so I'm not as concerned with which one of those is the actual default.

Or depending on what you mean: I think defaulting to any of the various time units would be fine. Here the existing default is similar to using `{"field": <insert field name>, "type": "temporal", "timeUnit": "yearmonthdatehoursminutesseconds"}` but as noted by Dominik, that isn't quite what happens.

My thoughts are:
- A default when not specifying the various properties should be somewhat useful by itself
- A default when not specifying the various properties ideally can be recreated manually by specifying properties, and my own bias would suggest that this should happen without having to set a ton of various properties

Of course this is all subjective. The current default may or may not fit these criteria, but I was surprised that it doesn't use a time unit at all (well, it sort of does). I'm also new-ish to vega-lite, so there could be other design constraints I'm totally missing (for example, maybe to your point it's just considered nonsense to try to plot temporal without a time transform).

**Reply by domoritz** [reply]
Thank you. I appreciate you thinking through this so deeply and caring about making a great user experience for everyone. 

There are cases where no time unit makes sense. For example, a line or point mark makes sense to plot without a time unit. It gets tricky when we have layered marks (e.g. Line and bar). 

Placing marks at the specified time like we do for linear scales still seems like a reasonable default.
